src/Arcus.BackgroundJobs.DataBricks/DatabricksInfoProvider.cs

```diff
@@ -0,0 +1,159 @@
+        /// </summary>
+        /// <param name="startOfWindow">The start of time window which we are interested in. (Inclusive)</param>
+        /// <param name="endOfWindow">The snd of time window which we are interested in. (Exclusive)</param>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the <paramref name="endOfWindow"/> is less than the <paramref name="startOfWindow"/>.</exception>
+        public async Task<IEnumerable<JobRun>> GetFinishedJobRunsAsync(DateTimeOffset startOfWindow, DateTimeOffset endOfWindow)
+        {
+            Guard.NotLessThan(endOfWindow, startOfWindow, nameof(endOfWindow), "Requires the date of the end window to be greater than the start date of the window");
+
+            IEnumerable<Run> jobRuns = await GetJobRunsAsync(startOfWindow, endOfWindow);
+            IEnumerable<Job> availableJobs = await _databricksClient.Jobs.List();
+
+            return from finishedJobRun in jobRuns
+                   where finishedJobRun.IsCompleted
+                   let jobName = availableJobs.FirstOrDefault(job => job.JobId == finishedJobRun.JobId)?.Settings?.Name
+                   select new JobRun(jobName, finishedJobRun);
+        }
+
+        private async Task<IEnumerable<Run>> GetJobRunsAsync(DateTimeOffset startOfWindow, DateTimeOffset endOfWindow)
+        {
+            var finishedJobs = new Collection<Run>();
+            bool hasMoreInformation;
+            var runOffset = 0;
+
+            do
+            {
+                RunList jobHistory = await _databricksClient.Jobs.RunsList(offset: runOffset);
+                if (jobHistory == null)
+                {
+                    break;
+                }
+
+                foreach (Run jobRun in jobHistory.Runs)
+                {
+                    if (jobRun.EndTime != null && jobRun.EndTime > startOfWindow && jobRun.EndTime < endOfWindow)
+                    {
+                        finishedJobs.Add(jobRun);
+                    }
+                    else
+                    {
+                        return finishedJobs;
+                    }
+                }
+
+                hasMoreInformation = jobHistory.HasMore;
+                runOffset += jobHistory.Runs.Count();
+            } while (hasMoreInformation);
+
+            return finishedJobs;
+        }
+
+        /// <summary>
+        /// Report a given <paramref name="jobRun"/> as a logging metric with the specified <paramref name="metricName"/>.
+        /// </summary>
+        /// <param name="metricName">The name of the logging metric.</param>
+        /// <param name="jobRun">The instance to report.</param>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="metricName"/> is blank.</exception>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="jobRun"/> is <c>null</c>.</exception>
+        public Task ReportFinishedJobOutcomeAsync(string metricName, JobRun jobRun)
```

Comment:
This and `MeasureJobOutcomesAsync` feel like a separate class that is using `DatabricksInfoProvider` because it doesn't provide info; thoughts?

`DatabricksMonitor` is an awful name but something in that direction?

Reply:
Moved to logger extension, so now it only contains Databricks-related stuff.